packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -770,9 +795,9 @@ class _RenderSegmentedButton<T> extends RenderBox with
 
   Size _calculateChildSize(BoxConstraints constraints) {
     double maxHeight = 0;
-    double childWidth = constraints.minWidth / childCount;
+    double childWidth = _isExpanded ? constraints.maxWidth : constraints.minWidth / childCount;
```

Comment:
I think it would be more readable here to completly seperate out the two code paths, e.g.

```dart
    double childWidth;
    if (_isExpanded) {
      childWidth = constraints.maxWidth / childCount;
    } else {
      childWidth = constraints.minWidth / childCount;
      RenderBox? child = firstChild;
      while (child != null) {
        childWidth = math.max(childWidth, child.getMaxIntrinsicWidth(double.infinity));
        child = childAfter(child);
      }
      childWidth = math.min(childWidth, constraints.maxWidth / childCount);
    }
```

Reply:
l got your point l applied this solution just l moved `RenderBox? child` outside of condition block because we are also using this variable for height calculation.